src/pages/index.js

```diff
@@ -4,91 +4,199 @@ import "../pages/index.css";
 import UserInfo from "../components/UserInfo.js";
 import Section from "../components/Section.js";
 import ImagePopup from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
 import FormPopup from "../components/PopupWithForm.js";
 import {
   initialCards,
   profileEditButton,
   profileEditModal,
   previewImageModal,
   addCardModal,
+  editAvatarModal,
+  popupConfirmModal,
   profileModalCloseButton,
   addCardModalCloseButton,
   previewImageCloseButton,
   addNewCardButton,
+  avatarEditSelector,
   profileTitle,
   profileDescription,
   profileTitleInput,
   profileDescriptionInput,
   previewImageFooter,
   profileEditForm,
   addCardForm,
+  editAvatarForm,
   cardTemplate,
   cardListEl,
   cardTitleInput,
   cardURLInput,
   validationSettings,
 } from "../utils/constants.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "58a7c735-ac44-49af-aeeb-e746dacd4b77",
+    "Content-Type": "application/json",
+  },
+});
+
+const popupConfirm = new PopupConfirm(popupConfirmModal);
+popupConfirm.setEventListeners();
+
+const user = new UserInfo(profileTitle, profileDescription, avatarEditSelector);
+console.log(user);
+
+const renderCard = (cardData) => {
+  const card = new Card(
+    {
+      cardData,
+      userId: userId,
+      handleCardClick: (imageData) => {
+        previewImage.open(imageData);
+      },
+      handleDeleteClick,
+      handleLikeClick,
+    },
+    cardTemplate
+  );
+  cardSection.addItem(card.getCardView());
+};
+
+function handleDeleteClick(item) {
+  popupConfirm.open();
+  popupConfirm.setSubmitAction(() => {
+    api
+      .deleteCard(item.getId())
+      .then(() => {
+        item.removeCard();
+        popupConfirm.changeBtnStatus(false);
+        popupConfirm.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeClick(card) {
+  if (!card._isLiked) {
+    api
+      .addCardLike(card._id)
+      .then(() => {
+        card.updateLikeStatus(true);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .removeCardLike(card._id)
+      .then(() => {
+        card.updateLikeStatus(false);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
 
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (cardData) => {
-      cardSection.addItem(createCard(cardData));
-    },
+    renderer: renderCard,
   },
   cardListEl
 );
 
-cardSection.renderItems();
+let userId = "";
 
-const user = new UserInfo({ profileTitle, profileDescription });
-console.log(user);
+// Loading initial section
+
+api.getAppInfo().then(([cards, userData]) => {
+  userId = userData._id;
```

Comment:
Needs correction: don't forget to update the initial user avatar

Reply:
Initial user avatar is updated using user.setAvatar